specification/src/chapter4.adoc

```diff
@@ -2,69 +2,897 @@
+
+|===
+
+For a static partition TEE, domain level granularity can be sufficient as device access within TEE and Normal domains is governed by TEE OS and the rich OS respectively. It can be implemented using IOPMP. Policy can be controlled by boot configuration, by a HW or FW RoT.
+
+For a virtualized TEE, IOMTT enforces supervisor domain level access rules (physical isolation). IOMMU enforces guest and TA level access rules (virtualization), supporting device assignment to a guest TEE or a TA.  
+
+NOTE: IOMTT can also be sufficient for protecting Root devices in the sense that M-mode can enforce that its own resources are never assigned to another domain. Use of IOPMP or similar still adds further protections. For example, a system may require that Root devices cannot be used to access memory assigned to Confidential domain.
+
+==== System integration
+
+In the case of a Global Platform TEE system a rich OS in Normal domain is free to schedule services, including TEE services, on any Hart available to it. The number and make-up of supervisor domains can be known, and a simple convention can be used for common identification (SDID value, see xref:chapter3.adoc#_supervisor_domains[supervisor domains]) of Normal, TEE, and Root domains across multiple Harts in a system. 
+
+System integration in this context involves providing _security attributes_ on a system interconnect, tagging all transactions (CPU or system agent initiated) to either Root, Normal, or TEE domains. 
+
+Possible use cases include:
+
+* Tweaking cryptographic memory protection (uniqueness)
+* Tagging interrupts, debug accesses, or coherent memory accesses
+* Device assignment (IOPMP/IOMTT integration), static or dynamic
+
+For interoperability, an implementation could adopt the following convention commonly used in Global Platform TEE ecosystems:
+
+[width=30%]
+[%header, cols="30,20"]
+|===
+| 
+| Security attribute
+
+| TEE domain
+| 2'b00
+
+| Normal domain
+| 2'b01
+
+| Root domain
+| 2'10
+
+| Reserved +
+(Confidential domain)
+| 2'11
+
+|===
+
+The attributes can be derived, for example, from SDID and privilege level, from PMA, or from MTT dynamic PMA (svpam).
+
+For some use cases security attributes can be extended to reflect finer granularity, for example for cryptographic memory protection with TA granularity.
+
+=== Debug and performance management
+
+See xref:chapter2.adoc#_security_lifecycle[security lifecycle]. +
+See https://github.com/riscv-non-isa/riscv-external-debug-security[enhanced RISC-V external debug security]
+
+[width=100%]
+[%header, cols="5,20"]
+|===
+| ID#     
+| Requirement
+
+| CAT_NNN
+| External debug MUST be enabled separately for Root domain.
+
+| CAT_NNN
+| External debug MUST be enabled separately for each supervisor domain.
+
+| CAT_NNN
+| External debug MUST only be enabled by a HW RoT (Root domain external debug) or by Root domain (supervisor domain external debug).
```

Comment:
HW RoT is optional elsewhere (should)

Reply:
In chapter two, I clarified the "MUST" and "SHOULD" for HW RoT.

Basically I don't think we can "mandate" HW RoT for every conceivable use case. Plenty of use cases, like "small" embedded systems, would be overkill to require a separate HW RoT. Whereas on "larger" complex systems it is a definite MUST.

I think the revised rules make this clearer.

In the use cases it is currently "SHOULD" in the generic "non-domain" case for the reasons above. And a MUST for GP TEE and CoVE use cases.